File: 2022/haskell/app/Day23.hs
module Day23 (day23) where


fp :: FilePath
fp = "../data/Day23.txt"


day23 :: IO ()
day23 = do
    contents <- readFile fp
    let
        part1 = "todo"
        part2 = "todo"
    putStrLn "Day 23:"
    putStrLn $ "Day 1: " ++ show part1
    putStrLn $ "Day 2: " ++ show part2

File: 2024/rust/src/day23.rs
static FILE_PATH: &'static str = "../data/day23.txt";

pub fn solutions() {
    let contents = std::fs::read_to_string(
        FILE_PATH
    ).expect("Broken File");
    println!("Day 23");
    part1(&contents);
    part2(&contents);
}

fn part1(contents: &str) -> usize {
    let answer = 0;
    // println!("Part 1: {}", answer);
    println!("Part 1: TODO");
    answer
}

fn part2(contents: &str) -> usize {
    let answer = 0;
    // println!("Part 2: {}", answer);
    println!("Part 2: TODO");
    answer
}

#[cfg(test)]
mod test {
    use super::{part1, part2};

    static DATA: &'static str = "
";

// #[test]
fn test_part1() {
    let answer = part1(DATA);
    assert_eq!(answer, 0);
}

// #[test]
fn test_part2() {
    let answer = part2(DATA);
    assert_eq!(answer, 0);
}

}
    